library/std/src/sys/pal/uefi/thread.rs

```diff
@@ -0,0 +1,50 @@
+use super::unsupported;
+use crate::ffi::CStr;
+use crate::io;
+use crate::num::NonZeroUsize;
+use crate::ptr::NonNull;
+use crate::time::Duration;
+
+pub struct Thread(!);
+
+pub const DEFAULT_MIN_STACK_SIZE: usize = 4096;
+
+impl Thread {
+    // unsafe: see thread::Builder::spawn_unchecked for safety requirements
+    pub unsafe fn new(_stack: usize, _p: Box<dyn FnOnce()>) -> io::Result<Thread> {
+        unsupported()
+    }
+
+    pub fn yield_now() {
+        // do nothing
+    }
+
+    pub fn set_name(_name: &CStr) {
+        // nope
+    }
+
+    pub fn sleep(dur: Duration) {
+        let boot_services: NonNull<r_efi::efi::BootServices> =
+            crate::os::uefi::env::boot_services().expect("can't sleep").cast();
+        let _ = unsafe { ((*boot_services.as_ptr()).stall)(dur.as_micros() as usize) };
```

Comment:
This seems wrong on 32-bit. Not that an hours-long sleep is sensible, but that's all it would take to overflow this, and the overflow behavior wouldn't be correct. (And I can *imagine* use cases for that.)

How about, if it's greater than usize::MAX, do a loop while that's the case, subtracting usize::MAX and sleeping that long?